### PR TITLE
[mds-cache] Add geospatial hashing to events/telemetry data

### PR DESCRIPTION
**Pull request by avatarneil**
## PR Checklist

 - [x] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [x] Provider
- [x] Agency
- [x] Audit
- [ ] Policy
- [ ] Compliance

### CODE REVIEW

File: packages/mds-audit/api.ts [hunk 1]
```diff
@@ -633,7 +632,7 @@ function api(app: express.Express): express.Express {
 
   app.get(pathsFor('/vehicles'), async (req, res) => {
     const { skip, take } = asPagingParams(req.query)
-    const bbox = getBoundingBox(JSON.parse(req.query.bbox))
+    const bbox = JSON.parse(req.query.bbox)
```
Comment (karcass) [inline packages/mds-audit/api.ts]: hm, this feels like inadequate input validation

File: packages/mds-cache/index.ts [hunk 2]
```diff
@@ -487,6 +526,26 @@ async function cleanup() {
   }
 }
 
+async function addGeospatialHash(key: string, coordinates: [number, number]) {
```
Comment (karcass) [inline packages/mds-cache/index.ts]: docs maybe?

File: packages/mds-cache/index.ts [hunk 3]
```diff
@@ -487,6 +526,26 @@ async function cleanup() {
...
+  const client = await getClient()
+  const [lat, lng] = coordinates
+  const res = await client.geoadd('locations', lng, lat, key)
+  return res
+}
+
+async function getEventsInBbox(bbox: BoundingBox) {
```
Comment (karcass) [inline packages/mds-cache/index.ts]: I've been using BBox not Bbox

File: packages/mds-logger/index.ts [hunk 4]
```diff
@@ -115,8 +115,10 @@ function makeCensoredLogMsgRecurse(msg: any): any {
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 function makeCensoredLogMsg(...msgs: any[]): any[] {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const censored = msgs.map(msg => makeCensoredLogMsgRecurse(msg))
-  return censored.map(msg => (typeof msg === 'object' ? JSON.stringify(msg) : msg))
+  return makeCensoredLogMsgRecurse(msgs).map((item: any) =>
```
Comment (karcass) [inline packages/mds-logger/index.ts]: why is this better?

File: packages/mds-audit/api.ts (hunk 1 above)
Comment (avatarneil) [reply packages/mds-audit/api.ts]: Meh, this is pretty much just app-facing at the moment so I'm not super concerned -- happy to write an isBBox function though

File: packages/mds-cache/index.ts (hunk 2 above)
Comment (avatarneil) [reply packages/mds-cache/index.ts]: sure

File: packages/mds-cache/index.ts (hunk 3 above)
Comment (avatarneil) [reply packages/mds-cache/index.ts]: kk, shall switch

File: packages/mds-logger/index.ts (hunk 4 above)
Comment (avatarneil) [reply packages/mds-logger/index.ts]: Current version no-does-worky properly, I noticed when tracing logs and fixed.

File: packages/mds-cache/index.ts [hunk 5]
```diff
@@ -46,23 +58,20 @@ declare module 'redis' {
     keysAsync: (arg1: string) => Promise<string[]>
     zaddAsync: (arg1: string | number, arg2: number, arg3: string) => Promise<number>
     zrangebyscoreAsync: (key: string, min: number | string, max: number | string) => Promise<string[]>
+    georadiusAsync: (
+      key: string,
+      longitude: number,
+      latitude: number,
+      radius: number,
+      unit: string
+    ) => Promise<string[]>
```
Comment (mdurling) [inline packages/mds-cache/index.ts]: `Promise<UUID[]>` for narrative purposes.

File: packages/mds-cache/index.ts [hunk 6]
```diff
@@ -122,6 +131,24 @@ async function hread(suffix: string, device_id: UUID): Promise<CachedItem> {
...
+  const [lat, lng] = coordinates
+  const res = await client.geoadd('locations', lng, lat, key)
+  return res
+}
+
+async function getEventsInBbox(bbox: BoundingBox) {
+  const client = await getClient()
+  const [pt1, pt2] = bbox
+  const [lng, lat] = [(pt1[0] + pt2[0]) / 2, (pt1[1] + pt2[1]) / 2]
+  const [radius, unit] = distance
```
Comment (mdurling) [inline packages/mds-cache/index.ts]: We have our own haversine function in utils ... perhaps we should remove it and use this everywhere.

File: packages/mds-cache/index.ts [hunk 7]
```diff
@@ -122,6 +131,24 @@ async function hread(suffix: string, device_id: UUID): Promise<CachedItem> {
   throw new Error(`${suffix} for ${device_id} not found`)
 }
 
+async function addGeospatialHash(key: string, coordinates: [number, number]) {
```
Comment (mdurling) [inline packages/mds-cache/index.ts]: `key: string` should be `device_id: UUID`

File: yarn.lock [hunk 8]
```diff
@@ -2946,6 +2946,11 @@ gauge@~2.7.3:
     strip-ansi "^3.0.1"
     wide-align "^1.1.0"
 
+geo-distance@0.2.0:
```
Comment (mdurling) [inline yarn.lock]: Odd that this is still hanging around.

**Pull request merged**